File: shared_resources/src/main/java/com/thewizrd/shared_resources/weatherdata/AirQualityProviderInterface.java
package com.thewizrd.shared_resources.weatherdata;

import com.thewizrd.shared_resources.locationdata.LocationData;
import com.thewizrd.shared_resources.utils.WeatherException;

public interface AirQualityProviderInterface {
    AirQuality getAirQualityData(LocationData location) throws WeatherException;
}

File: shared_resources/src/main/java/com/thewizrd/shared_resources/weatherdata/AirQualityProviderInterface.kt
package com.thewizrd.shared_resources.weatherdata

import com.thewizrd.shared_resources.locationdata.LocationData
import com.thewizrd.shared_resources.utils.WeatherException
import com.thewizrd.shared_resources.weatherdata.model.AirQualityData

interface AirQualityProviderInterface {
    @Throws(WeatherException::class)
    suspend fun getAirQualityData(location: LocationData): AirQualityData?
}